docs/FeatureManagement/Samples/NoFilters.tests.json

```diff
@@ -0,0 +1,50 @@
+[
+    {
+        "TestID": "NoFilters1",
+        "FeatureFlagName": "BooleanTrueFeatureFlag",
+        "ExpectedResult": true,
+        "Description": "An Enabled Feature Flag with no Filters."
+    },
```

Comment:
> I think we need to think carefully about the format of the testcase. Currently, we have a property called "Expected Result."
> 
> Now, this property is either true/false or "Exception."
> 
> If we want to introduce variants in the future, what should the expected result look like?
> 
> Once variants are introduced, we need to check both the variant and feature flag status.
> 
> Further, should we unify the exception which will be thrown in different FM libs? In this case, we can even specify the exception type in the testcases. After we release variant, there can be multiple kinds of exceptions. This can make sure our FM libs have the unified behavior.
> 
> Besides, we can add some testcases for invalid feature filter settings. For example, for time window filter, the start should be earlier than the end. This kind of validation for feature filter settings will become important after we release recurring time window. The setting will be much more complex.
> 
> I proposed the following shape for our test suites:
> 
> ```json
> {
>     "TestId": "ExampleTest-HelpfulNameForDebuging",
>     "FeatureFlagName": "ExampleFeatureFlag",
>     "Inputs": {}
>     "ExpectedResult":{
>        "FeatureStatus": true,
>        "Variant": "",
>        "Exception": ""
>     }
>     "Description": "1. Describe the feature flag and the input. 2. Describe the reason why it should get the expected result.",
> }
> ```
> 
> IMO, the test suite and samples we are implementing here is super valuable. It will be used by all FM libs in the future. Besides, the repo is open to our customer. Here can be the place for our customer to learn how to use feature management. We should put more efforts on it.

Good Point but, I think for "ExpectedResult"

{
  "IsEnabled": "true"/"false"/"exception" # This is a string
  "GetVariant": { # This would be "empty" for None/Null
    "name": "",
   "configuration": <This part may be tricky in some languages>
  }
}

Swapped the names to match the method names. Variants are more complex so it will need to be an object.

The exception part is tricky as we can say it will throw an exception, but the type will be different for each one. We could have the word exception be the value, but the message we throw could possibly be the same. Thoughts?

Reply:
```
{
  "IsEnabled": "true"/"false"/"exception" # This is a string
  "GetVariant": { # This would be "empty" for None/Null
    "name": "",
    "configuration": <This part may be tricky in some languages>
  }
}
```
I like this one. 

> The exception part is tricky as we can say it will throw an exception, but the type will be different for each one. We could have the word exception be the value, but the message we throw could possibly be the same. Thoughts?

I just realized that exception can from either the `IsEnabled` or `GetVariant` method. So I think we'd better have a separated property for exception.

How about this?

```javascript
"exception":{
    "hasException": true/false // this can be removed, since empty "exception" means no exception
    "message/type": "xxx",
    "when/from":"xxx"
}
```

Reply:
I'm good with your suggestion, though what do you mean by "when/from", do you have an example.

Reply:
"when/from" means which method this exception is from?

Does it happen in GetVariant or IsEnabled?

Reply:
> "when/from" means which method this exception is from?
> 
> Does it happen in GetVariant or IsEnabled?

The `is_enabled` and `get_variant` should have if it's an exception or not if we follow the above. Having her complicates things if both throw the exception.

Reply:
What will the exception from `get_variant` look like if we follow the *above*?

What do you mean by *above*? This one?

``` json
{
  "IsEnabled": "true"/"false"/"exception" # This is a string
  "GetVariant": { # This would be "empty" for None/Null
    "name": "",
    "configuration": <This part may be tricky in some languages>
  }
}
```

Reply:
> "IsEnabled": "true"/"false"/"exception" # This is a string

Sorry about the delayed response as I just get back to work. Maybe you've already agreed on this. But the "true"/"false" of string type looks a little bit confusing, esp. when the key name "IsEnabled" indicates it should be a boolean... 

If we need something to represent an exception, can it be `null`? Meaning that we cannot determine whether it's enabled or not because of the exception. I.e. "IsEnabled": true/false/null

AFAIU, the only benefit of using string would be the extensibility. Will there be more possible values for `IsEnabled`?